src/lib/swapper/swappers/ThorchainSwapper/getThorTradeQuote/getTradeQuote.ts

```diff
@@ -35,11 +39,16 @@ export type ThorEvmTradeQuote = TradeQuote &
 export type ThorTradeUtxoOrCosmosQuote = TradeQuote & ThorTradeQuoteSpecificMetadata
 export type ThorTradeQuote = ThorEvmTradeQuote | ThorTradeUtxoOrCosmosQuote
 
+export const isThorTradeQuote = (quote: TradeQuote | undefined): quote is ThorTradeQuote =>
+  !!quote && 'tradeType' in quote && quote.tradeType !== undefined
```

Comment:
And if we make it non-optional, then we don't need this anymore 
```suggestion
  !!quote && 'tradeType' in quote
```